v0/destinations/tiktok_ads/transform.js

```diff
@@ -0,0 +1,259 @@
+const { SHA256 } = require("crypto-js");
+const get = require("get-value");
+const { EventType } = require("../../../constants");
+const {
+  getErrorRespEvents,
+  CustomError,
+  constructPayload,
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  removeUndefinedAndNullValues,
+  defaultBatchRequestConfig,
+  getSuccessRespEvents,
+  isDefinedAndNotNullAndNotEmpty
+} = require("../../util");
+const {
+  trackMapping,
+  TRACK_ENDPOINT,
+  BATCH_ENDPOINT,
+  eventNameMapping,
+  MAX_BATCH_SIZE
+} = require("./config");
+
+function checkIfValidSHA256(str) {
+  // Regular expression to check if string is a SHA256 hash
+  const regexExp = /^[a-f0-9]{64}$/gi;
+
+  return regexExp.test(str);
+}
+
+function checkIfValidPhoneNumber(str) {
+  // Ref - https://ads.tiktok.com/marketing_api/docs?id=1701890979375106
+  // Regular expression to check whether it has country code
+  // but should not include +86
+  const regexExp = /^(\+(?!86)\d{1,3}){0,1}[0-9]{10}$/gi;
+
+  return regexExp.test(str);
+}
+
+const trackResponseBuilder = async (message, { Config }) => {
+  const pixel_code = Config.pixelCode;
+
+  let event = get(message, "event");
+  event = event ? event.trim().toLowerCase() : event;
+  if (!event) {
+    throw new CustomError("Event name is required", 400);
+  }
+
+  if (eventNameMapping[event] === undefined) {
+    throw new CustomError(`Event name (${event}) is not valid`, 400);
+  }
+  event = eventNameMapping[event];
+
+  let payload = constructPayload(message, trackMapping);
+  payload = { pixel_code, event, ...payload };
+
+  /*
+   * Hashing user related detail i.e external_id, email, phone_number
+   */
+
+  const external_id = payload.context.user.external_id.trim();
+  if (
+    isDefinedAndNotNullAndNotEmpty(external_id) &&
+    !checkIfValidSHA256(external_id)
+  ) {
+    payload.context.user.external_id = SHA256(external_id).toString();
+  }
+
+  const email = payload.context.user.email.trim().toLowerCase();
+  if (isDefinedAndNotNullAndNotEmpty(email) && !checkIfValidSHA256(email)) {
+    payload.context.user.email = SHA256(email).toString();
+  }
+
+  const phone_number = payload.context.user.phone_number.trim();
+  if (
+    isDefinedAndNotNullAndNotEmpty(phone_number) &&
+    !checkIfValidSHA256(phone_number)
+  ) {
+    if (checkIfValidPhoneNumber(phone_number)) {
+      payload.context.user.phone_number = SHA256(phone_number).toString();
+    } else {
+      throw new CustomError(
+        "Invalid phone number. Include proper country code except +86. Aborting ",
+        400
+      );
+    }
+  }
+
+  const response = defaultRequestConfig();
+  response.headers = {
+    "Access-Token": Config.accessToken,
+    "Content-Type": "application/json"
+  };
+
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.endpoint = TRACK_ENDPOINT;
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  return response;
+};
+
+const process = async event => {
+  const { message, destination } = event;
+
```

Comment:
Here you should add a check for all required config fields if they are empty strings or not.

Reply:
added this check. I'd removed this in my previous commit since it was driven from the web app as a req field but can fail when we provide manual input while passing to our transformer endpoint port:9090.

Reply:
One can send an empty string from config as well.

Reply:
This is already being taken care of. It throws an error not found.